packages/react-next/src/components/Shimmer/Shimmer.base.tsx

```diff
@@ -1,126 +1,94 @@
-  }
-
-  public componentDidUpdate(prevProps: IShimmerProps): void {
-    const { isDataLoaded } = this.props;
+  /**
+   * Flag for knowing when to remove the shimmerWrapper from the DOM.
+   */
+  const [contentLoaded, setContentLoaded] = React.useState(isDataLoaded);
 
-    if (isDataLoaded !== prevProps.isDataLoaded) {
-      this._async.clearTimeout(this._lastTimeoutId);
+  const divStyleProp = { width: width ? width : '100%' };
 
-      // Removing the shimmerWrapper div from the DOM only after the fade out animation completed.
+  React.useEffect(() => {
+    if (isDataLoaded !== contentLoaded) {
       if (isDataLoaded) {
-        this._lastTimeoutId = this._async.setTimeout(() => {
-          this.setState({
-            contentLoaded: isDataLoaded,
-          });
+        internalState.lastTimeoutId = setTimeout(() => {
+          setContentLoaded(true);
         }, TRANSITION_ANIMATION_INTERVAL);
+
+        return () => clearTimeout(internalState.lastTimeoutId);
       } else {
-        this.setState({
-          contentLoaded: isDataLoaded,
-        });
+        setContentLoaded(false);
       }
     }
-  }
-
-  public componentWillUnmount(): void {
-    this._async.dispose();
-  }
-
-  public render(): JSX.Element {
-    const {
-      styles,
-      shimmerElements,
-      children,
-      isDataLoaded,
-      width,
-      className,
-      customElementsGroup,
-      theme,
-      ariaLabel,
-      shimmerColors,
-    } = this.props;
-
-    const { contentLoaded } = this.state;
-
-    this._classNames = getClassNames(styles!, {
-      theme: theme!,
-      isDataLoaded,
-      className,
-      transitionAnimationInterval: TRANSITION_ANIMATION_INTERVAL,
-      shimmerColor: shimmerColors && shimmerColors.shimmer,
-      shimmerWaveColor: shimmerColors && shimmerColors.shimmerWave,
-    });
-
-    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
-
-    return (
-      <div {...divProps} className={this._classNames.root}>
-        {!contentLoaded && (
-          <div style={{ width: width ? width : '100%' }} className={this._classNames.shimmerWrapper}>
-            <div className={this._classNames.shimmerGradient} />
-            {customElementsGroup ? (
-              customElementsGroup
-            ) : (
-              <ShimmerElementsGroup
-                shimmerElements={shimmerElements}
-                backgroundColor={shimmerColors && shimmerColors.background}
-              />
-            )}
-          </div>
-        )}
-        {children && <div className={this._classNames.dataWrapper}>{children}</div>}
-        {ariaLabel && !isDataLoaded && (
-          <div role="status" aria-live="polite">
-            <DelayedRender>
-              <div className={this._classNames.screenReaderText}>{ariaLabel}</div>
-            </DelayedRender>
-          </div>
-        )}
-      </div>
-    );
-  }
-}
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
After thinking through this I'm pretty sure it's correct that you don't need to depend on `contentLoaded`, but can you please add a comment about why?